Source/Blazorise/Components/Highlighter/Highlighter.razor.cs

```diff
@@ -23,31 +26,33 @@ public partial class Highlighter : BaseComponent
     /// <inheritdoc/>
     protected override void OnParametersSet()
     {
-        fragments = GetFragments( Text, HighlightedText, CaseSensitive, UntilNextBoundary, NextBoundary );
+        allHighlightedTexts = new List<string>(HighlightedTexts ?? Enumerable.Empty<string>()) { HighlightedText };
+        fragments = GetFragments( Text,allHighlightedTexts, CaseSensitive, UntilNextBoundary, NextBoundary );
     }
 
-    private static IEnumerable<string> GetFragments( string text, string highlightedText, bool caseSensitive = false, bool untilNextBoundary = false, string nextBoundary = null )
+    static IEnumerable<Fragment> GetFragments( string text, List<string> highlightedTexts, bool caseSensitive = false, bool untilNextBoundary = false, string nextBoundary = null)
     {
-        if ( string.IsNullOrWhiteSpace( text ) )
-        {
-            return new List<string>();
-        }
-
-        if ( string.IsNullOrWhiteSpace( highlightedText ) )
-        {
-            return new List<string> { text };
-        }
-
-        highlightedText = Regex.Escape( highlightedText );
-
-        if ( untilNextBoundary )
-        {
-            highlightedText += nextBoundary;
-        }
-
-        return Regex
-            .Split( text, $"({highlightedText})", caseSensitive ? RegexOptions.None : RegexOptions.IgnoreCase )
-            .Where( s => !string.IsNullOrEmpty( s ) );
+        if (string.IsNullOrWhiteSpace(text))
+            return new List<Fragment>();
+
+        if (highlightedTexts == null || highlightedTexts.Count == 0 || highlightedTexts.All(string.IsNullOrWhiteSpace))
+            return new List<Fragment> { new(text, false) };
+
+        var escaped = highlightedTexts
+                      .Where(s => !string.IsNullOrWhiteSpace(s))
+                      .Select(Regex.Escape).ToList();
+
+        string pattern = untilNextBoundary
+          ? string.Join("|", escaped.Select(h => h + nextBoundary))
+          : string.Join("|", escaped);
+
+        
+        var regex = new Regex($"({pattern})", caseSensitive ? RegexOptions.None : RegexOptions.IgnoreCase);
```

Comment:
It's usually good to cache regex variable instead of recreating it every time.

Reply:
But how? It can change every time the method is called.

I can somehow create a dictionary with `(pattern,caseSensitive)` key, but I don't think it's worth it.